hr_attendance_reason/static/src/js/attendance_reason.js

```diff
@@ -0,0 +1,162 @@
+            delete context.params;
+            context.attendance_reason_id = parseInt(
+                this.$(".o_hr_attendance_reason").val(),
+                0
+            );
+            if (context.attendance_reason_id === 0) {
+                self.do_warn(_t("Please, select a reason"));
+            } else {
+                this._rpc({
+                    model: "hr.employee",
+                    method: "attendance_manual",
+                    args: [
+                        [self.employee.id],
+                        "hr_attendance.hr_attendance_action_my_attendances",
+                    ],
+                    context: context,
+                }).then(function(result) {
+                    if (result.action) {
+                        self.do_action(result.action);
+                    } else if (result.warning) {
+                        self.do_warn(result.warning);
+                    }
+                });
+            }
+        },
+    });
+    KioskConfirm.include({
+        events: _.extend(KioskConfirm.prototype.events, {
+            "click .o_hr_attendance_pin_pad_button_ok": _.debounce(
+                function() {
+                    this.update_attendance();
+                },
+                200,
+                true
+            ),
+        }),
+        start: function() {
+            var self = this;
+            this._rpc({
+                model: "res.company",
+                method: "search_read",
+                args: [
+                    [["id", "=", this.getSession().company_id]],
+                    ["show_reasons_on_attendance_screen"],
+                ],
+            }).then(function(res) {
+                self.show_reasons_on_attendance_screen =
+                    res[0].show_reasons_on_attendance_screen;
+            });
+            self.reasons = this.getAttendanceReasons();
+            this.getSession()
+                .user_has_group("hr_attendance.group_hr_attendance_use_pin")
+                .then(function(has_group) {
+                    self.use_pin = has_group;
+                    self.$el.html(
+                        QWeb.render("HrAttendanceKioskConfirm", {widget: self})
+                    );
+                    self.start_clock();
+                });
+            return self._super.apply(this, arguments);
+        },
+        getAttendanceReasons: function() {
+            var self = this;
+            this.reasons = [];
+            this._rpc({
+                model: "hr.attendance.reason",
+                method: "search_read",
+                fields: ["name", "action_type"],
+                domain: [["show_on_attendance_screen", "=", true]],
+            }).then(function(reasons) {
+                _.each(reasons, function(reason) {
+                    self.reasons.push(reason);
+                });
+            });
+            return this.reasons;
+        },
+        update_attendance: function() {
```

Comment:
This could be inherited as suggested in the other widget with `session.user_context`